### ISSUE TITLE
App 4.2.2.7 crashes on iOS 15.3.1

### ISSUE THREAD

**Issue by Cojnew** [nextcloud/ios#1916]
### Steps to reproduce

1. install app version 4.2.2.7 on iOS 15.3.1
2. open the app

### Expected behaviour

App should open.

### Actual behaviour

Splash screen is shown shortly, then app disappears again.
In the list of open apps, Nextcloud is still shown with the splash screen. When switching to it, same behaviour as when starting it.

### Screenshots

N/A, only splash screen is shown anyway.

### Logs

I will be happy to post any logs, but I cannot open the app... Any hints are welcome how to get any logs.

Settings > Nextcloud > Version shows "4.2.2.7"

This behaviour appeared today, and yesterday everything was still working fine.
I'd assume that the next behaviour was triggered by an update of the Nextcloud app, since I'm not aware of any change on server side.
I've got automatic updates of apps enabled, so I cannot tell when exactly the current version was installed.
In the app store version 4.2.2 is displayed as "1 week ago".

Behaviour seems identical to issue 1876, which was reported as fixed with 4.2.2 already...

### Reasoning or why should it be changed/implemented?

Crashing is bad.

### Environment data

**iOS version:** e.g. iOS 15.3.1

**Nextcloud iOS app version:** 4.2.2.7

**Server operating system:** Debian

**Web server:** Lighttpd

**Database:** mariadb

**PHP version:** PHP 7.0.33-0+deb9u12

**Nextcloud version:** 15.0.14 (yes, very old, but the app does not even get so far as to send any request to the server anyway...)

**Comment by thisIsTheFoxe** [comment]
Hi, thanks for the report.
If possible please go to Settings.app > Privacy > Analytics > Analytics Data and see if there is a crash report for the Nextcloud.app there. Also, feel free to try it with the latest [TestFlight](https://github.com/nextcloud/ios#testflight) version. 

> the app does not even get so far as to send any request to the server 

Did you verify that (i.e. check the server logs)? Because (afaik) the app can send requests even before the splash screen disappears :)

**Comment by Cojnew** [comment]
Hi, thanks for the prompt reply.

> If possible please go to Settings.app > Privacy > Analytics > Analytics Data and see if there is a crash report for the Nextcloud.app there.

There are some files called `Nextcloud-<timestamp>.ips`, but with timestamps from three months ago or older.
Thus, I'd assume that these reports are unrelated to the current problems. Should I post them anyway?
Should I look for other file names?

> Did you verify that (i.e. check the server logs)?

Yes, I was watching the lighttpd logs while opening the app, and there were no requests.

**Comment by Cojnew** [comment]
> Also, feel free to try it with the latest TestFlight version.

Update 1: I've tried all four versions currently available in TestFlight. and the issue seems to appear with all of them.
Versions tested: 4.2.0 (5), 4.2.1 (7), 4.2.2 (7), 4.3.0 (3)

> Did you verify that (i.e. check the server logs)?

Update 2: I checked the webserver logs for earlier requests. The last requests were two days ago, when the app was still working, but (surprise!) the client already identified itself as "Mozilla/5.0 (iOS) Nextcloud-iOS/4.2.2" from February 22, i.e. the current version. Thus, it looks like the app version update happened already ten days before the app started crashing...

If it wasn't the app version update, are there any chances to narrow down what could have triggered the behaviour change?

**Comment by Cojnew** [comment]
Update 4: Issue persisted with TestFlight version 4.3.0 (4).

Lacking other ideas, I suspected something in the app's data state to trigger the crash.
Since I could not live without the app any longer ;-), I uninstalled and re-installed the app, thus resetting its state to zero.
App did start up normally, asked for my login details, and I re-applied my few personal settings. Everything's fine.

It might or might not be possible any more now to find out the actual reason behind the crash.
Problem is solved for me, and as far as I'm concerned the issue can be closed...

**Comment by Cojnew** [comment]
> do you remember if you enabled the auto-upload feature? And if so, did you also enable the upload-in-background setting?

Yes, I remember: I had both the auto-upload feature and the upload-in-background setting enabled.

However, I'm not sure whether the upload-in-background setting works correctly on my phone:
New photos only lead to a red notification icon, if anything, but mostly are recognized only when I open the app.
Then, I still need to leave the app in the foreground for any uploads to finish, despite the "background" setting.

**Comment by madfusker** [comment]
I’m now experiencing the same issue on 4.2.7.  I do use the auto upload feature and it was working fine except for location change detection.  I’m guessing maybe the app auto updated and with the new version I’m now seeing the crash on open.  I suppose killing the app and reinstall is the only option?  The last time I did this I lost all my settings for auto upload.
